### PR TITLE
[CMake][Exp PyROOT] Keep PYTHON_VERSION_STRING with '.' and set DUNDE…

### PR DESCRIPTION
**Pull request by maxgalli**
…R_PYTHON_VERSION_STRING to build PyROOT instead

- Keep PYTHON_VERSION_STRING as set by CMake
- Set DUNDER_PYTHON_VERSION_STRING to build PyROOT
- Remove from Cache not needed OTHER... variables
- Revert 2082d7ce

Plus fixes in parsing of libcppyy_backend library name and extension, which caused:
http://cdash.cern.ch/viewTest.php?onlyfailed&buildid=839381

### CODE REVIEW

File: cmake/modules/SearchRootCoreDeps.cmake [hunk 1]
```diff
@@ -37,9 +37,10 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.14)
     set(PYTHON_EXECUTABLE "${Python3_EXECUTABLE}" CACHE INTERNAL "" FORCE)
     set(PYTHON_INCLUDE_DIRS "${Python3_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
     set(PYTHON_LIBRARIES "${Python3_LIBRARIES}" CACHE INTERNAL "" FORCE)
+    set(PYTHON_VERSION_STRING "${Python3_VERSION}" CACHE INTERNAL "" FORCE)
     set(PYTHON_VERSION_MAJOR "${Python3_VERSION_MAJOR}" CACHE INTERNAL "" FORCE)
     set(PYTHON_VERSION_MINOR "${Python3_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
-    set(PYTHON_VERSION_STRING "${Python3_VERSION_MAJOR}_${Python3_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
+    set(DUNDER_PYTHON_VERSION_STRING "${Python3_VERSION_MAJOR}_${Python3_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
```
Comment (etejedor) [inline cmake/modules/SearchRootCoreDeps.cmake]: Dunder is an abbreviation of "double underscore", but here there is only one, so I would pick another name. `PYTHON_UNDERSCORE_VERSION_STRING` ? A bit long but... can't think of anything better.

Comment (etejedor) [reply cmake/modules/SearchRootCoreDeps.cmake]: `PYTHON_UNDER_VERSION_STRING` ? @oshadura what do you think?

Comment (oshadura) [reply cmake/modules/SearchRootCoreDeps.cmake]: PYTHON_INSTALL_VERSION_STRING ?

Comment (oshadura) [reply cmake/modules/SearchRootCoreDeps.cmake]: we need definitelly to add description what does this variable do @maxgalli

**Comment by maxgalli** [comment]
@phsft-bot build

File: cmake/modules/SearchRootCoreDeps.cmake [hunk 2]
```diff
@@ -51,21 +52,23 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.14)
       set(PYTHON_EXECUTABLE "${Python2_EXECUTABLE}" CACHE INTERNAL "" FORCE)
       set(PYTHON_INCLUDE_DIRS "${Python2_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
       set(PYTHON_LIBRARIES "${Python2_LIBRARIES}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_VERSION_STRING "${Python2_VERSION}" CACHE INTERNAL "" FORCE)
       set(PYTHON_VERSION_MAJOR "${Python2_VERSION_MAJOR}" CACHE INTERNAL "" FORCE)
       set(PYTHON_VERSION_MINOR "${Python2_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
-      set(PYTHON_VERSION_STRING "${Python2_VERSION_MAJOR}_${Python2_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_UNDER_VERSION_STRING "${Python2_VERSION_MAJOR}_${Python2_VERSION_MINOR}")
```
Comment (oshadura) [inline cmake/modules/SearchRootCoreDeps.cmake]: Why not here to cache it?

Comment (maxgalli) [reply cmake/modules/SearchRootCoreDeps.cmake]: because we need it only internally when I run the loops in all the PyROOT CMakeLists.txt

Comment (oshadura) [reply cmake/modules/SearchRootCoreDeps.cmake]: I was worried if it is visible out side of cmake/modules/SearchRootCoreDeps.cmake.

Comment (maxgalli) [reply cmake/modules/SearchRootCoreDeps.cmake]: when I built everything was fine. Do you want me to put it back anyway?

Comment (oshadura) [reply cmake/modules/SearchRootCoreDeps.cmake]: but then why in line 43 and 102 it is done in different way? It is confusing...

Comment (maxgalli) [reply cmake/modules/SearchRootCoreDeps.cmake]: because I forgot to remove it :) just changed

**Pull request merged**